### PR TITLE
automatically allow non rust naming conventions

### PR DESCRIPTION
**Pull request by framlog**
+ related issue: #562 

I just added those attributes at the root mod. And I'm not sure whether it should be better if we could set this setting in `build.rs`.

### CODE REVIEW

File: tests/expectations/tests/call-conv-field.rs [hunk 1]
```diff
@@ -37,6 +37,6 @@ impl Default for JNINativeInterface_ {
     fn default() -> Self { unsafe { ::std::mem::zeroed() } }
 }
 extern "stdcall" {
-    #[link_name = "_bar@0"]
+    #[link_name = "bar@0"]
```
Comment (fitzgen) [inline tests/expectations/tests/call-conv-field.rs]: This change is failing in CI. @emilio do you know what's up with this?

**Comment by fitzgen** [comment]
> And I'm not sure whether it should be better if we could set this setting in build.rs.

We can wait and see if anyone cares enough about these warnings to file an issue requesting configurability :)

File: tests/expectations/tests/call-conv-field.rs (hunk 1 above)
Comment (emilio) [inline tests/expectations/tests/call-conv-field.rs]: Yeah, this is due to us not testing on OSX (where we remove an extra underscore from symbols that clang adds). Of course, this is cross-compiled so a `cfg` check is not enough. Let's just remove this change from the PR for now.

**Comment by emilio** [comment]
@fitzgen is this ready to land? Looks ok to me.

**Comment by fitzgen** [comment]
Yeah, I didn't get an email about the new commit for whatever reason.

@bors-servo r+

Thanks @framlog! If you'd like to continue hacking on bindgen, I can point you at some more things :) Thanks again!

**Comment by fitzgen** [comment]
I guess one potential issue is that this will only work if one has enabled C++ namespaces, otherwise we won't generate the module definition and its `allow` attributes.

**Comment by framlog** [comment]
@fitzgen ok, I'd like to do more work on it.

**Comment by fitzgen** [comment]
> @fitzgen ok, I'd like to do more work on it. thx

I'd suggest looking into how we prepend raw lines to the bindings, and inserting `#![allow(...)]` there. The raw lines get inserted regardless whether we are enabling C++ namespaces or not. Try grepping for `raw_lines` to get an idea what I'm talking about.

If anything doesn't make sense, don't hesitate to ask a question :)

**Comment by framlog** [comment]
@fitzgen Inserting an attribute at the beginning of a generated file is easy. However, I can't find a way to make the generated file can work in integration if I just insert that line. (Seems that rustc doesn't allow `include!` inner attributes in a mod) Could I generate a top mod if enabling C++ namespaces are not set?

**Comment by emilio** [comment]
> I'd be hesitant to allow that by default, it could break legit usages of `include!`.

**Comment by fitzgen** [comment]
Ah, good catch, I didn't think about how it would play out inside an `include!`.

**Comment by framlog** [comment]
Well, I managed to allow non rusty conventions by wrapping the whole generated code in a root mod regardless of enabling c++ namespaces. It doesn't bother the integration I think. And I could make a pr  after I fixing the name mangling issue if you think it's ok.

**Comment by emilio** [comment]
I guess that's fair, though that's a much more breaking change...

**Comment by framlog** [comment]
ok, thanks.

**Pull request merged**